components_library/boards/x86_board.py

```diff
@@ -94,10 +95,17 @@ def __init__(
                 "X86Motherboard will only work with the X86 ISA."
             )
 
+        if memory.get_size() > toMemorySize("3GB"):
+            raise Exception(
+                "X86Board currently only supports memory sizes up "
+                "to 3GB because of the I/O hole."
+            )
+        data_range = AddrRange(memory.get_size())
+        memory.set_memory_range([data_range])
+
         # Add the address range for the IO
-        # TODO: This should definitely NOT be hardcoded to 3GB
         self.mem_ranges = [
-            AddrRange(Addr("3GB")),  # All data
+            data_range,  # All data
             AddrRange(0xC0000000, size=0x100000),  # For I/0
```

Comment:
There's nothing stopping these ranges overlapping right now, right?

Reply:
There's a check to make sure data_range is < 3GB. That's the best we can do right now.